controllers/nonadmin_controller_test.go

```diff
@@ -0,0 +1,353 @@
+package controllers
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/pointer"
+
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+)
+
+const defaultNonAdminImage = "quay.io/konveyor/oadp-non-admin:latest"
+
+type ReconcileNonAdminControllerScenario struct {
+	namespace       string
+	dpa             string
+	errMessage      string
+	eventWords      []string
+	nonAdminEnabled bool
+	deployment      *appsv1.Deployment
+}
+
+func createTestDeployment(namespace string) *appsv1.Deployment {
```

Comment:
I would call this `createTestDeploymentNeedingUpdate` or something to signify that this deployment is not the final expected result.